### PR TITLE
Add support for xmit_hash_policy for bond interface

### PR DESCRIPTION
**Pull request by unrulyv**
See: https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/7/html/networking_guide/sec-using_channel_bonding

### CODE REVIEW

**Comment by thom311** [comment]
current kernel supports 5 modes:

[1] https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/include/uapi/linux/if_bonding.h?id=ae6088216ce4b99b3a4aaaccd2eb2dd40d473d42#n92
[2] https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/drivers/net/bonding/bond_options.c?id=ae6088216ce4b99b3a4aaaccd2eb2dd40d473d42#n98

Also, the patch only exposes the values as numbers, while kernel accepts both numbers and names. Compare to
```
                 ArgValidatorStr("mode", enum_values=ArgValidator_DictBond.VALID_MODES),
```
Here the role opts to only expose the names only. I think the role should be consistent here. And if it opts to only expose either the names or the numbers, the names seem more important. Also, because the numbers aren't actually documented in https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/Documentation/networking/bonding.txt?id=ae6088216ce4b99b3a4aaaccd2eb2dd40d473d42#n843

If anybody cares about the numbers, a new ArgValidator type could be added to handle numbers and names both.

File: .travis.yml [hunk 1]
```diff
@@ -4,8 +4,6 @@ dist: xenial
 language: python
 matrix:
   include:
-    - python: 2.6
```
Comment (tyll) [inline .travis.yml]: Why did you remove this? It is an unrelated change for the new bond policy.

Comment (unrulyv) [reply .travis.yml]: Hi @tyll , sorry that was me just mucking around as the py2.6 test was failing with PyYAML and deprecated version. I've fixed this with my latest change to this pull request.

**Comment by unrulyv** [comment]
> current kernel supports 5 modes:
> 
> [1] https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/include/uapi/linux/if_bonding.h?id=ae6088216ce4b99b3a4aaaccd2eb2dd40d473d42#n92
> [2] https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/drivers/net/bonding/bond_options.c?id=ae6088216ce4b99b3a4aaaccd2eb2dd40d473d42#n98
> 
> Also, the patch only exposes the values as numbers, while kernel accepts both numbers and names. Compare to
> 
> ```
>                  ArgValidatorStr("mode", enum_values=ArgValidator_DictBond.VALID_MODES),
> ```
> 
> Here the role opts to only expose the names only. I think the role should be consistent here. And if it opts to only expose either the names or the numbers, the names seem more important. Also, because the numbers aren't actually documented in https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/Documentation/networking/bonding.txt?id=ae6088216ce4b99b3a4aaaccd2eb2dd40d473d42#n843
> 
> If anybody cares about the numbers, a new ArgValidator type could be added to handle numbers and names both.

Great suggestion, I've amended my change to use just a big list of values including the names, as suggested by the RHEL doco.

**Comment by unrulyv** [comment]
@thom311 what do you think about the last update I made to this commit? Any suggestion for further changes?

**Comment by pcahyna** [comment]
@unrulyv thanks for the update! At first glance it seems to me that docs (README) needs updating, as noted by @tyll .

**Comment by thom311** [comment]
> @thom311 what do you think about the last update I made to this commit? Any suggestion for further changes?

Hi. It looks pretty good. Thank you.

Only, I would also add the two additional modes that (recent) kernel supports. See https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/drivers/net/bonding/bond_options.c?id=ae6088216ce4b99b3a4aaaccd2eb2dd40d473d42#n98
